pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -18,6 +18,7 @@ package v1alpha1
 
```

Comment:
Should we have a mutating webhook modify incoming builds that specify `Source` to become a build that specifies a single `Sources`? From then on we can assume/require builds don't have `Source` specified. Then in the future we can remove the ability to specify `Source` at all.

The downside of relying on a webhook is that some providers don't support them, so we'd have to duplicate the check in the controller. :(

Reply:
> Should we have a mutating webhook modify incoming builds that specify Source to become a build that specifies a single Sources?

Sounds like a great idea. 

>The downside of relying on a webhook is that some providers don't support them, so we'd have to duplicate the check in the controller. :(

How about continuing support for couple of minor versions before dropping support for `Source` in builds altogether? 

Do we need to address this in this PR?